core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -71,6 +82,161 @@ private[spark] class CoarseGrainedExecutorBackend(
+          }
+          ResourceRealCounts(execResourceInfo.count, taskCount.toLong)
+        }
+        if (realCounts.execCount < realCounts.taskCount) {
+          throw new SparkException(s"Executor resource: $rName, count: ${realCounts.execCount} " +
+            s"isn't large enough to meet task requirements of: ${realCounts.taskCount}")
+        }
+        // also make sure the executor resource count on start matches the
+        // spark.executor.resource configs specified by user
+        val userExecCountConfigName =
+          SPARK_EXECUTOR_RESOURCE_PREFIX + rName + SPARK_RESOURCE_COUNT_POSTFIX
+        val userExecConfigCount = env.conf.getOption(userExecCountConfigName).
+          getOrElse(throw new SparkException(s"Executor resource: $rName not specified " +
+            s"via config: $userExecCountConfigName, but required " +
+            s"by the task, please fix your configuration"))
+        val execConfigCountWithUnits = if (userExecConfigUnits.nonEmpty) {
+          val count = try {
+            Utils.byteStringAsBytes(userExecConfigCount + userExecConfigUnits.get)
+          } catch {
+            case e: NumberFormatException =>
+              // Ignore units not of byte types and just use count
+              logWarning(s"Illegal resource unit type, spark only " +
+                s"supports conversion of byte types, units: $userExecConfigUnits, " +
+                s"ignoring the type and using the raw count.", e)
+              userExecConfigCount.toLong
+          }
+          count
+        } else {
+          userExecConfigCount.toLong
+        }
+        if (execConfigCountWithUnits != realCounts.execCount) {
+          throw new SparkException(s"Executor resource: $rName, count: ${realCounts.execCount} " +
+            s"doesn't match what user requests for executor count: $execConfigCountWithUnits, " +
+            s"via $userExecCountConfigName")
+        }
+      } else {
+        throw new SparkException(s"Executor resource config missing required task resource: $rName")
+      }
+    }
+  }
+
+  // visible for testing
+  def parseResources(resourcesFile: Option[String]): Map[String, ResourceInformation] = {
+    // only parse the resources if a task requires them
+    val taskResourceConfigs = env.conf.getAllWithPrefix(SPARK_TASK_RESOURCE_PREFIX)
+    val resourceInfo = if (taskResourceConfigs.nonEmpty) {
+      val execResources = resourcesFile.map { resourceFileStr => {
+        val source = new BufferedInputStream(new FileInputStream(resourceFileStr))
+        val resourceMap = try {
+          val parsedJson = parse(source).asInstanceOf[JArray].arr
+          parsedJson.map(_.extract[ResourceInformation]).map(x => (x.name -> x)).toMap
+        } catch {
+          case e @ (_: MappingException | _: MismatchedInputException | _: ClassCastException) =>
+            throw new SparkException(
+              s"Exception parsing the resources passed in: $SPARK_TASK_RESOURCE_PREFIX", e)
```

Comment:
Should this be parsing the `resourceFileStr` ?

Reply:
yes